pdf_compression/models/ir_attachment.py

```diff
@@ -0,0 +1,171 @@
+        if values.get('mimetype') == 'application/pdf' and values.get(
+                'datas'):
+            try:
+                values['datas'] = self._run_ghostscript(values['datas'])
+            except (OSError, IOError, ValueError):
+                _logger.warning('You need Ghostscript to compress pdfs.')
+
+        return super(IrAttachment, self).write(values)
+
+    @api.model
+    def _run_ghostscript(self, pdf):
+        ir_values = self.env['ir.values']
+        compression = ir_values.get_default('ir.attachment', 'compression')
+        pdfa_option = ir_values.get_default('ir.attachment', 'pdfa_option')
+        report_override = self._context.get('pdf_option')
+
+        if pdfa_option != 'no' or report_override != 'default' or \
+                compression != 'no':
+
+            command_args = []
+            local_command_args = []
+
+            with NamedTemporaryFile(delete=False, suffix='.pdf') as f1:
+                f1.write(base64.decodestring(pdf))
+
+            f2 = NamedTemporaryFile(delete=False, suffix='.pdf')
+            f2.close()
+
+            input_file = f1.name
+            output_file = f2.name
+
+            if pdfa_option != 'no' and (report_override == 'default' or
+                                        not report_override):
+                if pdfa_option == 'pdfa1b':
+                    local_command_args.extend([
+                        '-dPDFA=1',
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                    ])
+                elif pdfa_option == 'pdfa2b':
+                    local_command_args.extend([
+                        '-dPDFA=2',
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                        '-dEmbedAllFonts=true',
+                        '-dMaxSubsetPct=100',
+                        '-dSubsetFonts=true'
+                    ])
+
+            if report_override != 'default':
+                if report_override == 'pdfa1b':
+                    local_command_args.extend([
+                        '-dPDFA=1',
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                    ])
+                elif report_override == 'pdfa2b':
+                    local_command_args.extend([
+                        '-dPDFA=2',
+                        '-dProcessColorModel=/DeviceRGB',
+                        '-dColorConversionStrategy=/RGB',
+                        '-dEmbedAllFonts=true',
+                        '-dMaxSubsetPct=100',
+                        '-dSubsetFonts=true'])
+
+            if compression == 'no' and (pdfa_option == 'pdfa1b' or
+                                        report_override == 'pdfa1b'):
+                local_command_args.extend(['-dPDFSETTINGS=/default'])
+            elif compression == 'prepress':
```

Comment:
These elif blocks could be simplified by using string interpolation for the settings:

```
if compression in ['prepress', ...]: 
    local_command_args.extend(['-dPDFSETTINGS=/%s' % compression])
```